compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1877,19 +1884,36 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 .map(|adt| adt.did())
                 != range_def_id
         {
-            let instead = self
+            // Suppress any range expr type mismatches
+            if let Some(mut diag) = self
+                .tcx
+                .sess
+                .diagnostic()
+                .steal_diagnostic(last_expr_field.span, StashKey::MaybeFruTypo)
+            {
+                diag.delay_as_bug();
+            }
+
+            let expr = self
                 .tcx
                 .sess
                 .source_map()
                 .span_to_snippet(range_end.expr.span)
-                .map(|s| format!(" from `{s}`"))
-                .unwrap_or_default();
-            err.span_suggestion(
-                range_start.span.shrink_to_hi(),
-                &format!("to set the remaining fields{instead}, separate the last named field with a comma"),
-                ",",
-                Applicability::MaybeIncorrect,
-            );
+                .ok()
+                .filter(|s| s.len() < 25 && !s.contains(|c: char| c.is_control()));
```

Comment:
As someone new to this part of the codebase, this line is unclear to me. What's `s`, and why do we filter it like this? Why 25?

Reply:
This is to avoid rendering very long expressions in "to set the remaining fields in `$EXPR`"

Reply:
25 is an arbitrary number.

Reply:
And `is_control` is to avoid rendering something like:

```
to set the remaining fields in `a
.b`
```

Reply:
Makes sense :) Perhaps leave a comment to this effect?

Reply:
Sure thing.